202305_coastal_ecosystems_TC/impact_calculation/exposure_ecosystem_matching.py

```diff
@@ -0,0 +1,78 @@
+import geopandas as gpd
+from climada.entity import Exposures
+import pandas as pd
+
+# This script creates an exposure file from vectorised WorldPop population data (per TC basin)
+# It also merges habitat data with the population data (4 permutations of two habitat datasets and two pop datasets)
+# The merged files can later be used to analyse impacts per habitat category
+
+# variables
+geo_epsg = 4326  # WGS84
+proj_eck4 = '+proj=eck4 +lon_0=0 +x_0=0 +y_0=0 +datum=WGS84 +units=m +no_defs'  # World Eckert IV
+prot_dist = 2000
+pop_years = ['2000', '2020']
+hab_years = ['1992', '2020']
+basins = ['AP', 'WP', 'IO', 'SH']
+output = '/cluster/work/climate/shuelsen/manuscript/results/intermediate/'
+
+# create exposure files for both years per basin
+for basin in basins:
+    for pop_year in pop_years:
+        # Load data
+        pop_path = f'/cluster/work/climate/shuelsen/manuscript/data/worldpop/wp_{pop_year}_{basin}.shp'
+        pop = gpd.read_file(pop_path)
+
+        # prepare dataframe
+        pop['longitude'] = pop['geometry'].x
+        pop['latitude'] = pop['geometry'].y
+
+        # write exposures
+        exp_pop = Exposures(pop)
+
+        # save exposure data
+        exp_pop.write_hdf5(f'{output}exposure/exp_wp_{basin}_{pop_year}.hdf5')
+
+# create merged habitat and population files per basin
+for basin in basins:
+    for pop_year in pop_years:
+        for hab_year in hab_years:
+            # Load data
+            rhab_path = f'/cluster/work/climate/shuelsen/manuscript/data/rhab/basin_clips/rhab_{hab_year}_{basin}.shp'
+            pop_path = f'/cluster/work/climate/shuelsen/manuscript/data/worldpop/wp_{pop_year}_{basin}.shp'
```

Comment:
These paths are not useful for an external user. Please make relative paths or define default variables where users can define their own paths.